ninjalooter/raidgroups.py

```diff
@@ -0,0 +1,812 @@
+import random
+import math
+
+from models import Player
+
+
+
+
+
+#
+# class for an EQ group
+#
+class Group():
+    
+    def __init__(self, group_type = 'General'):
+
+        # info to support assigning a score to this group reflecting how well it matches a target profile
+        self.MAX_SLOT               = 100   # score for having a slot filled with an exact match
+        self.MIN_SLOT               = 10    # score for having a slot filled with anyone
+        self.LEVEL_PENALTY          = 15    # per level less than 60
+        self.CLASS_PENALTY          = 50    # penalty if not an exact class match but using a substitute class
+        self.GENERAL_PENALTY        = 0.7   # penalize scores of players in general groups, to encourage their placement in the specific-role groups
+
+        # group info
+        self.group_type             = group_type
+        self.player_list            = []
+        self.group_score            = 0
+        self.max_group_score        = 0
+
+    def __repr__(self):
+        rv = '{:<10} Members: ({}): '.format(self.group_type, len(self.player_list))
+        for p in self.player_list:
+            rv += p.name + ', '
+        rv += '(SA score: {})'.format(self.max_group_score)
+        return rv
+
```

Comment:
pep8 wants one newline between functions in classes. lots of cleanup for that specific issue throughout this file I think :D